modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/dataset_display/DatasetDisplay.js

```diff
@@ -341,16 +340,17 @@ function DatasetDisplay({
 						value={selectedItemsValue.join(',')}
 					/>
 				)}
-				{items?.length > 0 && (
+				{items?.length > 0 ? (
```

Comment:
If `items` is `undefined`, then `items?.length` will be `undefined` too, which means that this is comparing a boolean with a number:

```js
undefined > false ? (...) : (...)
```

Now, `undefined > false` gives the result you want (`false`), so it's ok to leave it like this, but it is a bit sloppy. An alternative construction that compares numbers with numbers would be:

```js
items?.length ?? 0 ? (...) : (...)
```

(Yes, it's horrible, using three different operators that contain `?` in one expression. 🤦‍♂️)

Another construction would be nested ternaries. Some people don't like these; and when in JSX they are can get particularly ghastly:

```js
<div>
	{items ? (
		items.length > 0 ? (
			<CurrentViewComponent
				datasetDisplayContext={DatasetDisplayContext}
				items={items}
				itemsActions={itemsActions}
				style={style}
				{...currentViewProps}
			/>
		) : (
			<EmptyResultMessage />
		)
	) : (
		<EmptyResultMessage />
	)}
</div>
```

Reply:
> (Yes, it's horrible, using three different operators that contain ? in one expression. 🤦‍♂️)

I wish we could use the Spanish syntax for that kind of thing: `¿items.length?`. While that's not in ES2021, I'm gonna go with `items?.length ?? 0 ? (...) : (...)`. Good catch. Thanks!

Reply:
![spanishinquisition](https://user-images.githubusercontent.com/905006/86590106-33e38a00-bf8f-11ea-81b3-4b995398a062.gif)

Reply:
Here it is https://github.com/liferay-frontend/liferay-portal/pull/75/commits/b4bf64970125de7622f58e57713f4a09589aa5aa

Reply:
![good-one](https://user-images.githubusercontent.com/7074/86592780-62b02f00-bf94-11ea-9116-51e2dd80ea30.gif)